client/service/src/lib.rs

```diff
@@ -380,57 +380,35 @@ where
 		}
 	}
 
-	let (max_request_size, ws_max_response_size, http_max_response_size) =
-		legacy_cli_parsing(config);
-
 	let random_port = |mut addr: SocketAddr| {
 		addr.set_port(0);
 		addr
 	};
 
-	let ws_addr = config
-		.rpc_ws
+	let addr = config
+		.rpc_addr
 		.unwrap_or_else(|| "127.0.0.1:9944".parse().expect("valid sockaddr; qed"));
-	let ws_addr2 = random_port(ws_addr);
-
-	let http_addr = config
-		.rpc_http
-		.unwrap_or_else(|| "127.0.0.1:9933".parse().expect("valid sockaddr; qed"));
-	let http_addr2 = random_port(http_addr);
+	let addr2 = random_port(addr);
```

Comment:
I assume this is the "backup" address if the requested port is blocked? Some comment would be nice.